api/v1alpha1/gitsync_types.go

```diff
@@ -98,13 +98,13 @@ type Destination struct {
 // CommitStatus maintains the status of syncing an individual Git commit
 type CommitStatus struct {
 	// Hash of the git commit
-	Hash string `json:"hash"`
+	Hash string `json:"hash,omitempty"`
 
 	// Synced indicates if the sync went through
 	Synced bool `json:"synced,omitempty"`
 
 	// SyncTime represents the last time that we attempted to sync this commit (whether or not it succeeded)
-	SyncTime metav1.Time `json:"syncTime"`
+	SyncTime metav1.Time `json:"syncTime,omitempty"`
```

Comment:
Thanks for fixing but trying to understand what happened. I see our `CommitStatus` [field](https://github.com/numaproj-labs/numaplane/pull/97/files#diff-d43361aee7a4a6b3365a5445375072d8587995c112f648dd0eb8132549f1d391R67) itself is already "omitempty". So, do we assume that one got pushed but with these fields not set?

Reply:
yeah I think it is related to we take out the mapping to repo from `CommitStatus`. Now we only have one CommitStatus field, which may check inside what is the value there.

Reply:
oh, in that case I wonder if the `CommitStatus` should be a pointer? I wonder if that would take care of it? And then maybe we could still have fields in here that are mandatory if you do have a `CommitStatus`

Reply:
I was just going though how k8s handle this, so they handle the same ways as @juliev0  mentioned, like we have `CommitStatus` as pointer and inside that we have each field as omitempty. As example [StatusDetails](https://github.com/kubernetes/kubernetes/blob/91ee30074bee617d52fc24dc85132fe948aa5153/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L776) is used as pointer [here](https://github.com/kubernetes/kubernetes/blob/91ee30074bee617d52fc24dc85132fe948aa5153/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L764) with `omitempty`

Reply:
Sounds good!

Reply:
@chandankumar4 Although in your example all fields in `StatusDetails` is `omitempty`.